### PR TITLE
Fixed bug when using non generic visitor in HashSerializeEventSubscriber

### PR DESCRIPTION
**Pull request by turbo-ele**
| Q | A
| --- | ---
| Bug fix? | yes
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | none
| Related issues/PRs | none
| License | MIT
| Documentation PR | none

#### What's in this PR?

HashSerializeEventSubscriber called addData for every Visitor, even though it's only supported by GenericVisitor instances.

#### Why?

Because that was a bug ;)

### CODE REVIEW

File: phpunit.xml.dist [hunk 1]
```diff
@@ -2,11 +2,8 @@
 <phpunit bootstrap="./tests/bootstrap.php" colors="true">
 
     <testsuites>
-        <testsuite name="unit">
-            <directory suffix="Test.php">./src/Sulu/Component/*/Tests/Unit</directory>
-        </testsuite>
-        <testsuite name="functional">
-            <directory suffix="Test.php">./src/Sulu/Component/*/Tests/Functional</directory>
+        <testsuite name="components">
```
Comment (wachterjohannes) [inline phpunit.xml.dist]: why is that change necessary? i find it useful that first all unit tests run and the the "longer" running functional tests.

Comment (wachterjohannes) [inline phpunit.xml.dist]: as far as i see only the `Hash` component is not structured like that! it's ok for me to merge it but we should keep this structure although and move this tests into the Unit folder.

Only to say for bigger refactorings its interresting to only run the functional testcases to see if the function is ok and run the unit tests later.

Comment (danrot) [inline phpunit.xml.dist]: Ok, but I think that's something open to discuss. I think there should not be any functional tests in the component, because the components are only fully integrated in the bundles, and that's where functional testing should be happening.
